src/NuGet.Core/NuGet.Commands/CommandRunners/PushRunner.cs

```diff
@@ -62,17 +62,26 @@ public static class PushRunner
                 if (symbolPackageUpdateResource != null)
                 {
                     symbolSource = symbolPackageUpdateResource.SourceUri.AbsoluteUri;
-                    symbolApiKey = apiKey;
                 }
             }
 
+            // Precedence for package API key: -ApiKey param, config
+            apiKey ??= CommandRunnerUtility.GetApiKey(settings, packageUpdateResource.SourceUri.AbsoluteUri, source);
+
+            // Precedence for symbol package API key: -SymbolApiKey param, config, package API key
+            if (!string.IsNullOrEmpty(symbolSource))
+            {
+                symbolApiKey ??= CommandRunnerUtility.GetApiKey(settings, symbolSource, symbolSource);
+                symbolApiKey ??= apiKey;
```

Comment:
Thank you for your contribution. However, the proposed logic in this PR presents a security issue by potentially leaking the NuGet.org API key, as detailed in the following security advisory: [GHSA-3885-8gqc-3wpf](https://github.com/NuGet/NuGet.Client/security/advisories/GHSA-3885-8gqc-3wpf).

In my opinion, one possible solution is to verify whether the symbol source is obtained through a command line or a package feed URL. If the symbol source is obtained via a command line, then it is not secure to use the package source API key. On the other hand, if the symbol source is obtained through the package source index.json file, it may be safe to use the same API key.

Reply:
Thanks for your feedback.

I updated my changes according to your remarks.